pkg/operator/operator.go

```diff
@@ -218,5 +224,10 @@ func (optr *Operator) maoConfigFromInfrastructure() (*OperatorConfig, error) {
 			NodeLink:           machineAPIOperatorImage,
 			MachineHealthCheck: machineAPIOperatorImage,
 		},
+                BaremetalPodConfig: BaremetalConfig{
```

Comment:
Rather than make this empty image config, we should make this nil if platform != baremetal.

Reply:
Done.